org/qmckl_distance.org

```diff
@@ -775,6 +775,280 @@ end function test_qmckl_dist
+    - ~m > 0~
+    - ~n > 0~
+    - ~lda >= 3~ if ~transa == 'N'~
+    - ~lda >= m~ if ~transa == 'T'~
+    - ~ldb >= 3~ if ~transb == 'N'~
+    - ~ldb >= n~ if ~transb == 'T'~
+    - ~ldc >= m~
+    - ~A~ is allocated with at least $3 \times m \times 8$ bytes
+    - ~B~ is allocated with at least $3 \times n \times 8$ bytes
+    - ~C~ is allocated with at least $m \times n \times 8$ bytes
+
+*** C header
+
+    #+CALL: generate_c_header(table=qmckl_distance_rescaled_args,rettyp=get_value("CRetType"),fname=get_value("Name"))
+
+    #+RESULTS:
+    #+begin_src c :tangle (eval h_func) :comments org
+    qmckl_exit_code qmckl_distance_rescaled (
+          const qmckl_context context,
+          const char transa,
+          const char transb,
+          const int64_t m,
+          const int64_t n,
+          const double* A,
+          const int64_t lda,
+          const double* B,
+          const int64_t ldb,
+          double* const C,
+          const int64_t ldc,
+          const double  rescale_factor_kappa);
+    #+end_src
+
+*** Source
+    #+begin_src f90 :tangle (eval f)
+integer function qmckl_distance_rescaled_f(context, transa, transb, m, n, &
+     A, LDA, B, LDB, C, LDC, rescale_factor_kappa) &
+     result(info)
+  use qmckl
+  implicit none
+  integer(qmckl_context)  , intent(in)  :: context
+  character  , intent(in)  :: transa, transb
+  integer*8  , intent(in)  :: m, n
+  integer*8  , intent(in)  :: lda
+  real*8     , intent(in)  :: A(lda,*)
+  integer*8  , intent(in)  :: ldb
+  real*8     , intent(in)  :: B(ldb,*)
+  integer*8  , intent(in)  :: ldc
+  real*8     , intent(out) :: C(ldc,*)
+  real*8     , intent(in)  :: rescale_factor_kappa
+
+  integer*8 :: i,j
+  real*8    :: x, y, z, dist, rescale_factor_kappa_inv
+  integer   :: transab
+
+  rescale_factor_kappa_inv = 1.0d0/rescale_factor_kappa;
+
+  info = QMCKL_SUCCESS
+
+  if (context == QMCKL_NULL_CONTEXT) then
+     info = QMCKL_INVALID_CONTEXT
+     return
+  endif
+
+  if (m <= 0_8) then
+     info = QMCKL_INVALID_ARG_4
+     return
+  endif
+
+  if (n <= 0_8) then
+     info = QMCKL_INVALID_ARG_5
+     return
+  endif
+
+  if (transa == 'N' .or. transa == 'n') then
+     transab = 0
+  else if (transa == 'T' .or. transa == 't') then
+     transab = 1
+  else
+     transab = -100
+  endif
+
+  if (transb == 'N' .or. transb == 'n') then
+     continue
+  else if (transa == 'T' .or. transa == 't') then
+     transab = transab + 2
+  else
+     transab = -100
+  endif
+
+  if (transab < 0) then
+     info = QMCKL_INVALID_ARG_1
+     return
+  endif
+
+  if (iand(transab,1) == 0 .and. LDA < 3) then
+     info = QMCKL_INVALID_ARG_7
+     return
+  endif
+
+  if (iand(transab,1) == 1 .and. LDA < m) then
+     info = QMCKL_INVALID_ARG_7
+     return
+  endif
+
+  if (iand(transab,2) == 0 .and. LDA < 3) then
+     info = QMCKL_INVALID_ARG_7
+     return
+  endif
+
+  if (iand(transab,2) == 2 .and. LDA < m) then
+     info = QMCKL_INVALID_ARG_7
+     return
+  endif
+
```

Comment:
OK, Actually I was going to do this. The reason I didn't add this is because we cannot simply check for fabs(kappa) > 1.0e-7. These thresholds need to be consistent with the numerical precision variables - QMCKL_DEFAULT_PRECISION and QMCKL_DEFAULT_RANGE. I haven't figured out how to do this consistently.
Do you have an idea ?